src/CoreWCF.Primitives/src/CoreWCF/Channels/SecurityBindingElement.cs

```diff
@@ -0,0 +1,898 @@
+        {
+            this.GetSupportingTokensCapabilities(this.EndpointSupportingTokenParameters, out supportsClientAuth, out supportsWindowsIdentity);
+        }
+
+        // SecureConversation needs a demuxer below security to 1) demux between the security sessions and 2) demux the SCT issue and renewal messages
+        // to the authenticator
+        internal void AddDemuxerForSecureConversation(ChannelBuilder builder, BindingContext secureConversationBindingContext)
+        {
+            // add a demuxer element  right below security unless there's a demuxer already present below and the only 
+            // binding elements between security and the demuxer are "ancillary" binding elements like message encoding element and
+            // stream-security upgrade element. We could always add the channel demuxer below security but not doing so in the ancillary
+            // binding elements case improves perf
+            //int numChannelDemuxersBelowSecurity = 0;
+            //bool doesBindingHaveShapeChangingElements = false;
+            //for (int i = 0; i < builder.Binding.Elements.Count; ++i)
+            //{
+            //    if ((builder.Binding.Elements[i] is MessageEncodingBindingElement) || (builder.Binding.Elements[i] is StreamUpgradeBindingElement))
+            //    {
+            //        continue;
+            //    }
+            //    if (builder.Binding.Elements[i] is ChannelDemuxerBindingElement)
+            //    {
+            //        ++numChannelDemuxersBelowSecurity;
+            //    }
+            //    else if (builder.Binding.Elements[i] is TransportBindingElement)
+            //    {
+            //        break;
+            //    }
+            //    else
+            //    {
+            //        doesBindingHaveShapeChangingElements = true;
+            //    }
+            //}
+            //if (numChannelDemuxersBelowSecurity == 1 && !doesBindingHaveShapeChangingElements)
+            //{
+            //    return;
+            //}
+
+            //ChannelDemuxerBindingElement demuxer = new ChannelDemuxerBindingElement(false);
+            //demuxer.MaxPendingSessions = this.LocalServiceSettings.MaxPendingSessions;
+            //demuxer.PeekTimeout = this.LocalServiceSettings.NegotiationTimeout;
+
+            //builder.Binding.Elements.Insert(0, demuxer);
+            //secureConversationBindingContext.RemainingBindingElements.Insert(0, demuxer);
+        }
+
+        internal void ApplyPropertiesOnDemuxer(ChannelBuilder builder, BindingContext context)
+        {
+            /* TODO later
+             Collection<ChannelDemuxerBindingElement> demuxerElements = builder.Binding.Elements.FindAll<ChannelDemuxerBindingElement>();
+             foreach (ChannelDemuxerBindingElement element in demuxerElements)
+             {
+                 if (element != null)
+                 {
+                     element.MaxPendingSessions = this.LocalServiceSettings.MaxPendingSessions;
```

Comment:
We need some throttling mechanism which we're still waiting on feature support in ASP.NET Core. Can you open an issue to add support to the new demuxer to support MaxPendingSessions

Reply:
https://github.com/CoreWCF/CoreWCF/issues/253